qiskit/ignis/measurement/discriminator/iq_discrimination.py

```diff
@@ -0,0 +1,191 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+import numpy as np
+from sklearn.discriminant_analysis import LinearDiscriminantAnalysis
+from sklearn.discriminant_analysis import QuadraticDiscriminantAnalysis
+
+from qiskit.ignis.characterization.fitters import BaseFitter
+from qiskit.exceptions import QiskitError
+from qiskit.result.models import ExperimentResult
+from qiskit.result import postprocess
+
+
+class IQDiscriminationFitter(BaseFitter):
+
+    def __init__(self, cal_results, qubits, circuit_names, expected_states,
+                 discriminant,  description):
+        """
+        Args:
+            cal_results: calibration results, list of qiskit.Result or
+            qiskit.Result
+            qubits: the qubits for which we want to discriminate.
+            circuit_names: The names of the circuits in cal_results.
+            expected_states: a list that should have the same length as
+                cal_results. If cal_results is a Result and not a list then
+                expected_states should be a string or a float or an int.
+            discriminant: a discriminant from e.g. scikit learn.
```

Comment:
I think this is tying the base class too closely too tightly to Scipy. I think it would be better to have this as an inheriting class such as `ScipyIQDiscriminationFitter`.

Reply:
You mean sklearn? I'm fine with the name change. Is there any need to generalize this beyond sklearn?

Reply:
One thing to keep in mind, the naming is getting heavy: `QuadraticScikitIQDiscriminationFitter`.

Reply:
This is true, I suppose this could be handled with docstring rather than in the name.

Reply:
The other option would be to create an `IQDiscriminationFitter` and leave `fit_fun` and `discriminate` as abstract methods (remove `discriminant` as a parameter for the base fitter), you could then implement these separately (or create a `ScikitIQDiscriminationFitter`) that implements these methods. I think this might be the best of both worlds.